lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -216,6 +206,13 @@ impl ShardReplicaSet {
         //   Check that shard snapshot is compatible with the collection
         //   (see `VectorsConfig::check_compatible_with_segment_config`)
 
+        let _partial_snapshot_search_lock = match recovery_type {
+            RecoveryType::Full => None,
+            RecoveryType::Partial => {
+                Some(self.partial_snapshot_meta.take_search_write_lock().await)
+            }
+        };
```

Comment:
Why not do the same on full recovery?

Reply:
Fair enough. I get that we assume use cases for full/partial recovery are different. Partial recoveries are ongoing and should not interrupt searches. While we don't expect full recoveries to be common, and they may block reads.